test/SettingsTest.php

```diff
@@ -0,0 +1,135 @@
+    {
+        parent::tearDown();
+        unset($this->settings);
+    }
+
+    /**
+     * @dataProvider settingsDataProvider
+     * @param $settingsData
+     */
+    public function testToArray($settingsData)
+    {
+        $this->assertEquals($this->settings->toArray(), $settingsData);
+    }
+
+    /**
+     * @dataProvider settingsDataProvider
+     * @param $settingsData
+     */
+    public function testGetUuid($settingsData)
+    {
+        $emptySettings = new Settings('', '', '', '', '');
+        $this->assertFalse($emptySettings->getUuid());
+        $this->assertEquals($this->settings->getUuid(), $settingsData['uuid']);
+    }
+
+    /**
+     * @dataProvider settingsDataProvider
+     * @param $settingsData
+     */
+    public function testGetWebhook($settingsData)
+    {
+        $this->assertEquals($this->settings->getWebhook('http://example1.com/webhooks'), $settingsData['webhook']['http://example1.com/webhooks']);
+        $this->assertEquals($this->settings->getWebhook('http://example2.com/webhooks'), $settingsData['webhook']['http://example2.com/webhooks']);
+        $this->assertFalse($this->settings->getWebhook('http://non_existing_url'));
+    }
+
+    /**
+     * @dataProvider settingsDataProvider
+     * @param $settingsData
+     */
+    public function testGetUrl($settingsData)
+    {
+        $this->assertEquals($this->settings->getUrl(), $settingsData['url']);
+    }
+
+    /**
+     * @dataProvider settingsDataProvider
+     * @param $settingsData
+     */
+    public function testGetApiKey($settingsData)
+    {
+        $this->assertEquals($this->settings->getApiKey(), $settingsData['apiKey']);
+    }
+
+    /**
+     * @dataProvider settingsDataProvider
+     * @param $settingsData
+     */
+    public function testGetSecretKey($settingsData)
+    {
+        $this->assertEquals($this->settings->getSecretKey(), $settingsData['secretKey']);
+    }
+
+    /**
+     * @dataProvider settingsDataProvider
+     * @param $settingsData
+     */
+    public function testSharedSecret($settingsData)
+    {
+        $this->assertEquals($this->settings->getSharedSecret(), $settingsData['sharedSecret']);
+    }
+
+    public function getSettingsData()
+    {
+        return [
+            'data' => [
+                'name' => 'testName',
+                'uuid' => '11111111-00000000-00000000-00000000',
+                'apiKey' => 'AAAAAA-AAAAAA-AAAAAA',
+                'secretKey' => 'BBBBBB-BBBBBB-BBBBBB',
+                'url' => 'https://test.url',
+                'sharedSecret' => null,
+                'webhook' => [
+                        'http://example1.com/webhooks' => '00000000-0000-0000-0000-000000000000',
+                        'http://example2.com/webhooks' => '11111111-0000-0000-0000-000000000000',
+                        ],
+                ],
+        ];
+    }
+
+    public function settingsDataProvider()
+    {
+        return [
+            $this->getSettingsData()
+        ];
+    }
+
```

Comment:
My brain has an obsession towards simplicity, so I am just telling if there is a case for having simpler code, let's take the chance and do it!

Reply:
Because then it will be needed to use this dataProvider like this in setUp():
```
$settingsParameters = $this->getSettingsData()[0];
```
Which in my opinion is less ugly to use additional function.

Reply:
Or duplicate the data from getSettingsData()

Reply:
if you return an array of data in your provider, you definitely need to use index `[0]`, in which case, I . agree that `['data']` is way better. But if your provider simply returns the required data, I don't think you need to use either of them.

Reply:
Provider should return an array of arrays.

Reply:
Sorry, I meant `getSettingsData` not the provider.

Reply:
I commited that already in response to your previous comment.